tests/components/mqtt/test_fan.py

```diff
@@ -1133,7 +1126,7 @@ async def test_sending_mqtt_commands_and_explicit_optimistic(hass, mqtt_mock, ca
         await common.async_turn_on(hass, "fan.test", preset_mode="auto")
 
     await common.async_turn_on(hass, "fan.test", preset_mode="whoosh")
-    assert mqtt_mock.async_publish.call_count == 2
+    assert mqtt_mock.async_publish.call_count == 3
```

Comment:
Why does this change?

Reply:
The fan platform used to raise `NotValidPresetModeError` in the compat shim which prevented the `async_turn_on` `ON` topic call from being sent. Now that the check is in the preset mode, the `ON` topic gets sent.

Before
```
[call('command-topic', 'ON', 0, False),
 call('preset-mode-command-topic', 'whoosh', 0, False)]
```

After
```
[call('command-topic', 'ON', 0, False),
 call('command-topic', 'ON', 0, False),
 call('preset-mode-command-topic', 'whoosh', 0, False)]
```

Reply:
Ah, ok. I suggest we reset the mock after that call as they aren't related to decrease confusion.

Reply:
4951822d2020c99e2a2e18cae20552f53f53c550